### PR TITLE
Move TE cross entropy guard to training args

### PR DESCRIPTION
**Pull request by yaoyu-33**
## Summary

Follow-up to #5115. This keeps the TE cross entropy fusion safety guard in `validate_args`, but removes it from `ModelParallelConfig.__post_init__` so programmatic config construction is not blocked.

- Allow `ModelParallelConfig(cross_entropy_loss_fusion=True, cross_entropy_fusion_impl='te')` to be constructed.
- Keep the training CLI / args validation assertion for the unsafe combination.
- Update unit coverage to reflect that split: core config can represent the setting, training args reject it.

## Test Plan

- `UV_CACHE_DIR=/home/yuya/Projects/Megatron-LM/.uv-cache uv run isort --check-only tests/unit_tests/test_model_parallel_config.py`
- `UV_CACHE_DIR=/home/yuya/Projects/Megatron-LM/.uv-cache uv run black --check tests/unit_tests/test_model_parallel_config.py megatron/core/model_parallel_config.py`
- `PYTHONPYCACHEPREFIX=/home/yuya/Projects/Megatron-LM/.pycache-check /home/yuya/mypython/bin/python -m py_compile megatron/core/model_parallel_config.py tests/unit_tests/test_model_parallel_config.py`

Local focused pytest could not complete on this workstation: direct run is blocked by the local `nvidia-resiliency-ext` dev version assertion (`0.6.0.dev69` compares below required `0.6.0`); masking NVRx as unavailable gets past that but the local CUDA/PyTorch driver mismatch segfaults during import.

### CODE REVIEW

**Review by cuichenx — APPROVED** [review]

**Review by deepakn94 — APPROVED** [review]

**Comment by ko3n1g** [comment]
/ok to test [288795f](https://github.com/NVIDIA/Megatron-LM/pull/5162/commits/288795f15eab38753080a80ea82571ac36ed322e)
